### DISCUSSION TITLE
Changing the installation path for an existing app does nothing

### DISCUSSION THREAD

**Discussion by KishoreT1** [wixtoolset/issues#6842] [Questions]
Please check the below details and provide your valuable suggestions. Thanks alot!

ProductVersion: 3.10

Visual Studio 2017

ToolsVersion="4.0"

.NET Version: 4.6.2

Windows Version: Windows Server 2016 Standard

Problem Details:

The "Change" Option on the installation wizard of the application.exe should allow me to specify a different folder / path for an already installed application. This is not happening.

Steps to reproduce:

Given I have an application installed
When I run the application.exe again, and select the "Change" option
And specify a different installation path, it does nothing

Current Results:

Nothing happens. Wizard completes, no errors. But the new specified folder contains nothing and the previous installation folder contains all files.

Expected Results:

Files should move from previous installation folder to the new one

#Bundle.wxs:

<?xml version="1.0" encoding="UTF-8"?>

<?define UpgradeCode = "E6423185-E6E9-4398-8E12-97846270F7FB" ?>
<?define RegistryKey = "SOFTWARE\Company1\Product1" ?>

<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi"
     xmlns:netfx="http://schemas.microsoft.com/wix/NetFxExtension"
     xmlns:util="http://schemas.microsoft.com/wix/UtilExtension"
     xmlns:bal="http://schemas.microsoft.com/wix/BalExtension">
    <Bundle Name="!(bind.packageName.connector)" 
          Version="!(bind.packageVersion.connector)" 
          Manufacturer="!(bind.packageManufacturer.connector)" 
          UpgradeCode="$(var.UpgradeCode)"
          IconSourceFile="$(var.IcoPath)"
          DisableModify="yes">

        
        <WixVariable Id="WixMbaPrereqPackageId" Value="Netfx462Full" />
        <WixVariable Id="WixMbaPrereqLicenseUrl" Value="NetfxLicense.rtf" />
        <WixVariable Id="PreqbaLogo" Value="$(var.PreqbaLogoPath)" />
    
        
        <Variable Name="UIStringFiles" Type="string" Value="WixUI.wxl;$(var.StringsWxl)"/>
        
        <BootstrapperApplicationRef Id="ManagedBootstrapperApplicationHost" >
          <Payload SourceFile="$(var.OutputPath)\InstallerBootstrapperUI.dll"/>
          <Payload SourceFile="$(var.OutputPath)\ConnectorBootstrapperUI.dll"/>
          <Payload SourceFile="$(var.OutputPath)\BootstrapperCore.dll"/>
          <Payload SourceFile="$(var.OutputPath)\BootstrapperCore.config" />
          <Payload SourceFile="$(var.OutputPath)\Microsoft.Deployment.WindowsInstaller.dll" />
          <Payload SourceFile="$(var.WindowsPatchDirectory)\Windows6.1-KB2999226-x64.msu" />
          <Payload SourceFile="$(var.WindowsPatchDirectory)\Windows8.1-KB2999226-x64.msu" />
          <Payload SourceFile="$(var.WindowsPatchDirectory)\Windows8-RT-KB2999226-x64.msu" />
          <Payload SourceFile="WixUI.wxl" />
          <Payload SourceFile="$(var.StringsWxl)" />
        </BootstrapperApplicationRef>

        <Variable Name="MainPackageId" Type="string" Value="connector" />
        <Variable Name="LaunchConfig" Type="numeric" Value="1" Persisted="yes" />
        <Variable Name="LaunchConfigExecutable" Type="string" Value="[InstallFolder]/ProxyUI.exe" />
        <Variable Name="LaunchConfigParam" Type="string" Value="/setup"/>
        <Variable Name="LaunchConfigParamRepair" Type="string" Value="/setup"/>
        <Variable Name="LaunchConfigElevated" Type="numeric" Value="1" />
        <Variable Name="InstallFolder" Type="string" Value="[ProgramFiles64Folder][DIR_PROG_COMPANY]\[DIR_PROG_PRODUCT]" Persisted="yes"/>
        <Variable Name="DisplayProductName" Type="string" Value="[PRODUCT_NAME]"/>
        <Variable Name="DisplayBrandProductName" Type="string" Value="[BRAND_PRODUCT_NAME]"/>
        <Variable Name="DisplayProductVersion" Type="string" Value="!(bind.packageVersion.connector)"/>
        <Variable Name="InstallFolderRegistrySearchKey" Type="string" Value="$(var.RegistryKey)" />
        <Variable Name="InstallFolderRegistrySearchValue" Type="string" Value="(Default)" /> 
        <Variable Name="IcoPath" Type="string" Value="$(var.IcoPath)" Persisted="yes" />
        <Variable Name="ProductName" Type="string" Value="$(var.ProductName)" Persisted="yes" />
        <Variable Name="InstallPackageErrorKB2999226_Windows81_x64" Type="string" Value="!(loc.INSTALL_KB2999226_ERROR)" Persisted="yes" />
        <Variable Name="InstallPackageErrorKB2999226_Windows8_x64" Type="string" Value="!(loc.INSTALL_KB2999226_ERROR)" Persisted="yes" />
        <Variable Name="InstallPackageErrorKB2999226_Windows61_x64" Type="string" Value="!(loc.INSTALL_KB2999226_ERROR)" Persisted="yes" />
        <Variable Name="InstallPackageErrorNetfx462Full" Type="string" Value="!(loc.INSTALL_DOTNET_4_6_2_ERROR)" Persisted="yes" />
        <Variable Name="InstallingPackageKB2999226_Windows81_x64" Type="string" Value="!(loc.INSTALL_KB2999226)" Persisted="yes" />
        <Variable Name="InstallingPackageKB2999226_Windows8_x64" Type="string" Value="!(loc.INSTALL_KB2999226)" Persisted="yes" />
        <Variable Name="InstallingPackageKB2999226_Windows61_x64" Type="string" Value="!(loc.INSTALL_KB2999226)" Persisted="yes" /> 
        <Variable Name="InstallingPackageNetfx462Full" Type="string" Value="!(loc.INSTALL_DOTNET_4_6_2)" Persisted="yes" />

        <util:RegistrySearch Root="HKLM" Key="SOFTWARE\Microsoft\Net Framework Setup\NDP\v4\Full" Value="Release" Variable="Netfx4FullVersion" />
        <util:RegistrySearch Root="HKLM" Key="SOFTWARE\Microsoft\Net Framework Setup\NDP\v4\Full" Value="Release" Variable="Netfx4x64FullVersion" Win64="yes" />
        <util:RegistrySearch Root="HKLM" Key="SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\Packages\Package_for_KB2919355~31bf3856ad364e35~amd64~~6.3.1.14" Value="CurrentState" Variable="KB2919355Installed" Win64="yes" />
        <util:FileSearch Id="WindowsUCRTVersion" Path="[SystemFolder]\api-ms-win-crt-runtime-l1-1-0.dll" Variable="WindowsUCRTVersion" Result="version" />
        <util:FileSearch Id="WindowsUCRTVersionExists" Path="[SystemFolder]\api-ms-win-crt-runtime-l1-1-0.dll" Variable="WindowsUCRTVersionExists" Result="exists" />

        <bal:Condition Message="Setup cannot continue because Windows Server 2008 R2 SP1 or above is required to install the $(var.ProductName)">
            <![CDATA[VersionNT64 > v6.1 OR (VersionNT64 = v6.1 AND ServicePackLevel >= 1)]]>
        </bal:Condition>
        
        <bal:Condition Message="The update corresponding to KB2919355 needs to be installed before you can install this product on Windows Server 2012 R2">
            <![CDATA[VersionNT64 <> v6.3 OR (VersionNT64 = v6.3 AND KB2919355Installed = 112)]]>
        </bal:Condition>
        <Chain>
          
          
          <ExePackage
              Id="EnableNetFx4"
              SourceFile="run.cmd"
              Compressed="yes"
              PerMachine="yes"
              DisplayName="!(loc.ENABLE_DOTNET_4_FEATURES)"
              LogPathVariable="EnableNetFx4Log"
              Permanent="yes"
              InstallCommand="[WindowsFolder]sysnative\dism.exe /online /enable-feature /featurename:NetFx4 /all &gt; [TempFolder]dism_netfx4_command.txt"
              DetectCondition="Netfx4FullVersion AND (NOT VersionNT64 OR Netfx4x64FullVersion)"
              InstallCondition="VersionNT &gt;= v6.2"
              bal:PrereqSupportPackage="yes"
              />
              
          
          <MsuPackage
            Id="KB2999226_Windows81_x64"
            SourceFile="$(var.WindowsPatchDirectory)\Windows8.1-KB2999226-x64.msu"
            InstallSize="970803"
            Permanent="yes"
            Vital="yes"
            DisplayName="!(loc.INSTALL_KB2999226)"
            InstallCondition="VersionNT = v6.3 AND VersionNT64"
            DetectCondition="WindowsUCRTVersionExists AND WindowsUCRTVersion &gt;= v10.0.10137.0"
            bal:PrereqSupportPackage="yes"
            KB="2999226">
          </MsuPackage>
          
          
          <MsuPackage
            Id="KB2999226_Windows8_x64"
            SourceFile="$(var.WindowsPatchDirectory)\Windows8-RT-KB2999226-x64.msu"
            InstallSize="1362211"
            Permanent="yes"
            Vital="yes" 
            DisplayName="!(loc.INSTALL_KB2999226)"
            InstallCondition="VersionNT = v6.2 AND VersionNT64"
            DetectCondition="WindowsUCRTVersionExists AND WindowsUCRTVersion &gt;= v10.0.10137.0"
            bal:PrereqSupportPackage="yes"
            KB="2999226">
          </MsuPackage>
          
          
          <MsuPackage
            Id="KB2999226_Windows61_x64"
            SourceFile="$(var.WindowsPatchDirectory)\Windows6.1-KB2999226-x64.msu"
            InstallSize="1012025"
            Permanent="yes"
            Vital="yes"
            DisplayName="!(loc.INSTALL_KB2999226)"
            InstallCondition="VersionNT = v6.1 AND VersionNT64"
            DetectCondition="WindowsUCRTVersionExists AND WindowsUCRTVersion &gt;= v10.0.10137.0"
            bal:PrereqSupportPackage="yes"
            KB="2999226">
          </MsuPackage>

          <ExePackage
              Id="Netfx462Full"
              InstallCommand="/q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx462FullLog].html&quot;"
              RepairCommand="/q /norestart /repair /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx462FullLog].html&quot;"
              UninstallCommand="/uninstall /q /norestart /ChainingPackage &quot;[WixBundleName]&quot; /log &quot;[NetFx462FullLog].html&quot;"
              PerMachine="yes"
              DetectCondition="Netfx4FullVersion &gt;= 394802 AND (NOT VersionNT64 OR Netfx4x64FullVersion &gt;= 394802)"
              DisplayName="!(loc.INSTALL_DOTNET_4_6_2)"
              Vital="yes"
              Permanent="yes"
              LogPathVariable="NetFx462FullLog"
              Compressed="yes"
              Protocol="netfx4"
              SourceFile="$(var.DotNetPackage)"
              bal:PrereqSupportPackage="yes">
              <ExitCode Behavior="forceReboot" Value="1641" />
              <ExitCode Behavior="forceReboot" Value="3010" />
              
              <ExitCode Behavior="forceReboot" Value="-2147021886" />
              <ExitCode Behavior="success" Value="0" />
              <ExitCode Behavior="error"/>
          </ExePackage>

          <MsiPackage
              Id="connector"
              SourceFile="$(var.OutputPath)\$(var.CultureGroupOutputFolder)\installer.msi"
              Compressed="yes"
              EnableFeatureSelection="yes"
              Vital="yes">
            <MsiProperty Name="INSTALLDIR" Value="[InstallFolder]" />
            <MsiProperty Name="ONPREM" Value="$(var.OnPrem)" />
          </MsiPackage>
          
        </Chain>
    </Bundle>
</Wix>

**Comment by robmen** [comment]
IIRC, the Windows Installer does not let you change the installation folder of Components during a repair/maintenance mode.
